bin/scoop.ps1

```diff
@@ -1,32 +1,56 @@
 #Requires -Version 5
-param($cmd)
+param($SubCommand)
 
-Set-StrictMode -off
+Set-StrictMode -Off
 
 . "$PSScriptRoot\..\lib\core.ps1"
 . "$PSScriptRoot\..\lib\buckets.ps1"
 . "$PSScriptRoot\..\lib\commands.ps1"
+. "$PSScriptRoot\..\lib\help.ps1"
+
 # for aliases where there's a local function, re-alias so the function takes precedence
 $aliases = Get-Alias | Where-Object { $_.Options -notmatch 'ReadOnly|AllScope' } | ForEach-Object { $_.Name }
 Get-ChildItem Function: | Where-Object -Property Name -In -Value $aliases | ForEach-Object {
     Set-Alias -Name $_.Name -Value Local:$($_.Name) -Scope Script
 }
 
-$commands = commands
-if ('--version' -contains $cmd -or (!$cmd -and '-v' -contains $args)) {
-    Write-Host "Current Scoop version:"
-    Invoke-Expression "git -C '$(versiondir 'scoop' 'current')' --no-pager log --oneline HEAD -n 1"
-    Write-Host ""
+switch ($SubCommand) {
+    ({ $SubCommand -in @($null, '--help', '/?') }) {
+        if (!$SubCommand -and $Args -eq '-v') {
+            $SubCommand = '--version'
+        } else {
+            exec 'help'
+        }
+    }
+    ({ $SubCommand -eq '--version' }) {
+        Write-Host 'Current Scoop version:'
+        if ((Test-CommandAvailable git) -and (Test-Path "$PSScriptRoot\..\.git") -and (get_config SCOOP_BRANCH 'master') -ne 'master') {
+            Invoke-Expression "git -C '$PSScriptRoot\..' --no-pager log --oneline HEAD -n 1"
+        } else {
+            $version = Select-String -Pattern '^## \[(v[\d.]+)\].*?([\d-]+)$' -Path "$PSScriptRoot\..\CHANGELOG.md"
+            Write-Host $version.Matches.Groups[1].Value -ForegroundColor Cyan -NoNewline
+            Write-Host " - Released at $($version.Matches.Groups[2].Value)"
+        }
+        Write-Host ''
 
-    Get-LocalBucket | ForEach-Object {
-        $bucketLoc =  Find-BucketDirectory $_ -Root
-        if(Test-Path (Join-Path $bucketLoc '.git')) {
-            Write-Host "'$_' bucket:"
-            Invoke-Expression "git -C '$bucketLoc' --no-pager log --oneline HEAD -n 1"
-            Write-Host ""
+        Get-LocalBucket | ForEach-Object {
+            $bucketLoc = Find-BucketDirectory $_ -Root
+            if (Test-Path (Join-Path $bucketLoc '.git')) {
```

Comment:
```suggestion
            if ((Test-Path (Join-Path $bucketLoc '.git')) -and (Test-CommandAvailable git)) {
```

Reply:
If there's `.git` subfolder, then the user must have had `git` installed, so I don't test `git`. Or may user pull the core package and buckets and then remove git?

Reply:
You're right, it's unlikely. I just put it there to be on the safe side. Otherwise this PR LGTM!

Reply:
Okay, add it and I'll merge the PR.